### PR TITLE
test: new UpdateEligibility unit tests

### PR DESCRIPTION
**Pull request by andrewmorris43**
## Description

Adds new unit tests for the UpdateEligibility function.
https://nhsd-jira.digital.nhs.uk/browse/DTOSS-2448

## Type of changes

- [x] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [x] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

**Comment by Mahifernando** [comment]
## Refrain from merging

Please refrain from merging this PR upon approval until the DevOps pipelines (PR validation) are configured to run your tests using the following command:

```
make test-unit
```

Engineers should also be able to execute the same command to run their unit tests locally.

File: application/CohortManager/src/Functions/ParticipantManagementServices/updateEligibility/updateEligibility.cs [hunk 1]
```diff
@@ -9,22 +9,22 @@
 
 namespace updateEligibility
```
Comment (Warren-Pitterson) [inline application/CohortManager/src/Functions/ParticipantManagementServices/updateEligibility/updateEligibility.cs:10]: Can this be moved to line 1 please, Can it also follow the format of ` <Organisation>.<Project>.<Function>.<Service>` 

e.g. NHS.CohortManager.CaasIntegration.UpdateEligibility

**Review by krca1 — APPROVED** [review]

File: tests/ParticipantManagementServicesTests/updateEligibilityTests/updateEligibilityTests.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+using Common;
+using System.Net;
+using System.Text.Json;
+using Model;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using System.Text;
+using updateEligibility;
+
+namespace NHS.CohortManager.Tests.ParticipantManagementService;
```
[the remark this answers is not in the document]
Comment (andrewmorris43) [reply tests/ParticipantManagementServicesTests/updateEligibilityTests/updateEligibilityTests.cs:12]: Yes will do. I obviously got the wrong end of the stick on this. Should the namespace always be on line 1? Just for future reference.

[the remark this answers is not in the document]
Comment (Warren-Pitterson) [reply tests/ParticipantManagementServicesTests/updateEligibilityTests/updateEligibilityTests.cs:12]: yes please, you're not the only one. i'm also used to having the namespace where you've put it but the NHS projects want them at the top

File: application/CohortManager/src/Functions/ParticipantManagementServices/updateEligibility/updateEligibility.cs [hunk 3]
```diff
@@ -7,59 +9,57 @@
...
-
-            }
-            catch (Exception ex)
-            {
-                _logger.LogInformation($"Unable to call function.\nMessage:{ex.Message}\nStack Trace: {ex.StackTrace}");
-            }
+        }
+        catch (Exception ex)
+        {
+            _logger.LogInformation($"Unable to call function.\nMessage:{ex.Message}\nStack Trace: {ex.StackTrace}");
```
Comment (SamRobinson75684) [inline application/CohortManager/src/Functions/ParticipantManagementServices/updateEligibility/updateEligibility.cs:58]: I don't think there's any point in having new lines in this log

File: tests/ParticipantManagementServicesTests/updateEligibilityTests/updateEligibilityTests.cs [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        _callFunction.Setup(call => call.SendPost(It.Is<string>(s => s.Contains("markParticipantAsEligible")), It.IsAny<string>()))
+                .Returns(Task.FromResult(_webResponse.Object));
+
+        SetupRequest(json);
+        var sut = new UpdateEligibility(_mockLogger.Object, _createResponse.Object, _callFunction.Object);
+
+        await sut.Run(_request.Object);
+
+        _createResponse.Verify(response => response.CreateHttpResponse(HttpStatusCode.OK, It.IsAny<HttpRequestData>()), Times.Once);
+        _createResponse.VerifyNoOtherCalls();
```
Comment (SamRobinson75684) [inline tests/ParticipantManagementServicesTests/updateEligibilityTests/updateEligibilityTests.cs:54]: in each one of these tests there's no assert, I think we should add an asset as well as the verify

File: application/CohortManager/src/Functions/ParticipantManagementServices/updateEligibility/updateEligibility.cs (hunk 3 above)
Comment (andrewmorris43) [reply application/CohortManager/src/Functions/ParticipantManagementServices/updateEligibility/updateEligibility.cs:58]: I'll remove them cheers

File: tests/ParticipantManagementServicesTests/updateEligibilityTests/updateEligibilityTests.cs (hunk 4 above)
Comment (andrewmorris43) [reply tests/ParticipantManagementServicesTests/updateEligibilityTests/updateEligibilityTests.cs:64]: I've added in an assert for each of the tests now.

**Review by SamRobinson75684 — APPROVED** [review]

**Pull request merged**